1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,47 @@
-function setAlarm() {}
+let globSetinterval;
+let isPaused = false;
+let timer;
+let min;
+let sec;
+
+function setAlarm() {
+  let timeRemaining = document.getElementById("timeRemaining");
+  let alarmSet = document.getElementById("alarmSet").value *60;
+    if (alarmSet > 0) {
+      timer = setInterval(() =>{
+
+    if (!isPaused){
+      alarmSet -- ;
+      let min = Math.floor(alarmSet/60);
+      let sec = alarmSet % 60;
```

Comment:
Did you discuss anything about scope in class Yousef.  Line 15 and 16 you have declared two variables and given them values. What about the variables declared at 4 and 5.  They are undefined.